expression/builtin_time_vec.go

```diff
@@ -1147,13 +1146,6 @@ func (b *builtinStrToDateDurationSig) vecEvalDuration(input *chunk.Chunk, result
 			result.SetNull(i, true)
 			continue
 		}
-		if hasNoZeroDateMode && (t.Year() == 0 || t.Month() == 0 || t.Day() == 0) {
-			if err := handleInvalidTimeError(b.ctx, types.ErrWrongValue.GenWithStackByArgs(types.DateTimeStr, t.String())); err != nil {
-				return err
-			}
-			result.SetNull(i, true)
-			continue
-		}
```

Comment:
Looks like non-vector mode has already fix this?

Reply:
yes. The non-vector mode ignores the value of hasNoZeroDateMode.

Reply:
> Looks like non-vector mode has already fix this?

Yes, the non-vector mode ignores the value of hasNoZeroDateMode.